File: Array/2053. Kth Distinct String in an Array/KthDistinct.cpp
#include <vector>
#include <string>
#include <unordered_map>

using namespace std;

class Solution {
public:
    string kthDistinct(vector<string>& arr, int k) {
        unordered_map<string, int> dict;

        // 统计每个字符串的出现次数
        for (const string& str : arr) {
            dict[str]++;
        }

        int count = 0;
        // 找到第k个独特的字符串
        for (const string& str : arr) {
            if (dict[str] == 1) {
                count++;
                if (count == k) {
                    return str;
                }
            }
        }
        return "";
    }
};

int main() {
    Solution solution;
    vector<string> arr = {"a", "b", "a", "c", "b"};
    int k = 2;
    cout << solution.kthDistinct(arr, k) << endl; // 输出 "c"
    return 0;
}

File: Array/2053. Kth Distinct String in an Array/KthDistinct.rs
use std::collections::HashMap;

struct Solution;

impl Solution {
    pub fn kth_distinct(arr: Vec<String>, k: i32) -> String {
        let mut dict: HashMap<String, i32> = HashMap::new();

        for str in &arr {
            let count = dict.entry(str.clone()).or_insert(0);
            *count += 1;
        }

        let mut count = 0;
        for str in &arr {
            if let Some(&1) = dict.get(str) {
                count += 1;
                if count == k {
                    return str.clone();
                }
            }
        }
        String::new()
    }
}

fn main() {
    let arr = vec!["a".to_string(), "b".to_string(), "a".to_string(), "c".to_string(), "b".to_string()];
    let k = 2;
    println!("{}", Solution::kth_distinct(arr, k)); // 输出 "c"
}